arm-documentdb/2015-04-08/swagger/documentdb.json

```diff
@@ -336,24 +336,18 @@
           }
         ],
         "responses": {
-          "200": {
-            "description": "The access key was regenerated successfully.",
-            "schema": {
-              "$ref": "#/definitions/DatabaseAccountListKeysResult"
-            }
+          "202": {
+            "description": "Accepted. The regenerate key operation will complete asynchronously."
           }
         }
       }
     },
-    "/subscriptions/{subscriptionId}/providers/Microsoft.DocumentDB/databaseAccountNames/{accountName}": {
+    "/providers/Microsoft.DocumentDB/databaseAccountNames/{accountName}": {
```

Comment:
Almost all the RPs have "/subscriptions/{subscriptionId}/providers/Microsoft.Storage/checkNameAvailability" have this format of the endpoint and it is a POST request.

However, docdb has implemented it differently as a HEAD request. Implementing it as a HEAD is fine. However over here you are checking whether a resource exists or not. A resource is uniquely identified with /subscriptions/{subscriptionId}/ ... So not having this wouldnt  be an antipattern for a Resource URI?